sdk/messaging/azeventhubs/event_data_batch.go

```diff
@@ -77,15 +77,14 @@ func (mb *EventDataBatch) toAMQPMessage() *amqp.Message {
 	mb.batchEnvelope.Format = batchMessageFormat
 
 	if mb.partitionKey != nil {
-		mb.batchEnvelope.Annotations[partitionKeyAnnotation] = *mb.partitionKey
-	}
+		if mb.batchEnvelope.Annotations == nil {
+			mb.batchEnvelope.Annotations = make(amqp.Annotations)
+		}
 
-	if mb.partitionID != nil {
```

Comment:
Why was this removed?

Reply:
It turns out it's not needed. 

We already route the batch sending to the proper link if the user explicitly chooses a partition for the batch _or_ we send it to the general link for hub otherwise (and then there'd be no partition id).